tests/conftest.py

```diff
@@ -138,17 +138,21 @@ def runtime_grpc_server(
         training_service=sample_train_service,
     ) as server:
 
-        grpc_thread = threading.Thread(
-            target=server.start,
-        )
-        grpc_thread.daemon = False
-        grpc_thread.start()
-        _check_server_readiness(server)
```

Comment:
This was a fun bug, the context manager already boots the server so this was just spinning up a thread that would immediately crash by trying to boot the server again